wormhole/file_transport.go

```diff
@@ -191,27 +191,24 @@ func (t *fileTransport) connectViaRelay(otherTransit *transitMsg) (net.Conn, err
 	failChan := make(chan string)
 
 	var count int
-
+	fmt.Println(otherTransit.HintsV1)
 	for _, relay := range otherTransit.HintsV1 {
 		if relay.Type == "relay-v1" {
 			for _, endpoint := range relay.Hints {
 				var addr string
 				switch endpoint.Type {
 				case "direct-tcp-v1":
-					addr = net.JoinHostPort(endpoint.Hostname, strconv.Itoa(endpoint.Port))
 					t.relayURL.Scheme = "tcp"
+					t.relayURL.Host = net.JoinHostPort(endpoint.Hostname, strconv.Itoa(endpoint.Port))
```

Comment:
Why are we overwriting `t.relayURL` here instead of creating a local variable? I can't see any direct harm this code may cause, but also not its purpose. So if this is intentional, maybe add a comment with the rationale for future reference

Reply:
Local variable would be better, thanks for noting. However, I am thinking now, theoretically, there can be several transit hints available (direct-tcp-1, websocket-1 ws, websocket-1 wss). Should we handle such cases? Now connectViaRelay loops via all hints, however connectToRelay switch to single what is set in relayUrl. Long term vision could be, like you get several urls, like websocket-1 in EU, websocket-1 in US, and you connect via first fastest or preferable or at least loop through all which you are able to connect.

Reply:
The rust implementation connects to all of them (with a bit of delay as to not spam connections) and then takes the first one that works

Reply:
It connects also to all, somehow overlooked at first.
Made adjustment